agent/app/agent.go

```diff
@@ -108,6 +110,26 @@ func newAgent(
 		return nil, err
 	}
 
+	var metadataManager containermetadata.MetadataManager
+	if cfg.ContainerMetadataEnabled {
+		// Get the oldest Docker Client version with up to date inspect API
+		// as some metadata is unavailable in older API client versions
+		// If this version is unavailable we use default
+		// Due to limitations of interfaces and import cycles
```

Comment:
I think your only import cycle issue is with the `containermetadata.ContainerMetadataClientVersion`.  I'd be in favor of moving this into the client/factory code directly and just moving the constant over there as well so you don't have an import cycle; that'll keep the impact localized to just the code that needs to be refactored anyway.